modules/assim.sequential/R/sda.enkf_refactored.R

```diff
@@ -350,19 +350,24 @@ sda.enkf <- function(settings,
     }
     
     #----chaning the extension of nc files to a more specific date related name
-    purrr::walk(
-      list.files(
-        path = file.path(settings$outdir, "out"),
-        "*.nc$",
-        recursive = TRUE,
-        full.names = TRUE),
+   files <-  list.files(
+      path = file.path(settings$outdir, "out"),
+      "*.nc$",
+      recursive = TRUE,
+      full.names = TRUE)
+   files <-  files[grep(pattern = "SDA*", files, invert = TRUE)]
+    
+    
+    purrr::walk(files,
       function(.x){
         file.rename(.x ,
                     file.path(dirname(.x),
-                              paste0(gsub(" ", "", as.character(names(obs.mean)[t])),
+                              paste0("SDA_",gsub(" ", "", as.character(names(obs.mean)[t])),
```

Comment:
Are all files in `files` guaranteed to live in different directories? If any of them share a `dirname`, they'll be given the same name and therefore overwrite each other. Or am I missing something?

Also, less importantly: `file.rename` accepts vectors, and `names()` is documented to return characters if it returns anything, so I *think* you can drop both `walk` and `as.character` and simplify this to

```
file.rename(
  files,
  file.path(
    dirname(files),
    paste0("SDA_", gsub(" ", "", names(obs.mean)[t]), ".nc"))
```

Reply:
I haven't seen any of our models to produce more than one `nc` output file per dir, but that's a good point and we can slightly modify that to use the `basename` but then we need the `walk` again something like this:
`paste0("SDA_", basename(.x), "_", names(obs.mean)[t], ".nc")`
>

Reply:
oh seems like the `basename` function is also vectorized and yeah seems like we don't the `walk` at all.

Reply:
@kzarada could you please give a try to this replacing the whole `walk` see if it works as expected :
`
file.rename(
    files,
    file.path(
       dirname(files),
       paste0("SDA_", basename(files), "_", gsub(" ", "", names(obs.mean)[t]), ".nc")
     )
 )
`

Reply:
That code worked- the output files look like this 'SDA_2019.nc_2019-11-0500:00:00.nc'

Reply:
I pushed the update